packages/core/src/conversation/ConversationService/ConversationService.ts

```diff
@@ -403,4 +406,48 @@ export class ConversationService {
   public async wipeMLSConversation(groupId: Uint8Array): Promise<void> {
     return this.mlsService.wipeConversation(groupId);
   }
+
+  public async handleEpochMismatch() {
+    this.logger.info(`There were some missed messages, handling possible epoch mismatch in MLS conversations.`);
+
+    //fetch all the mls conversations from backend
+    const conversations = await this.apiClient.api.conversation.getConversationList();
+    const foundConversations = conversations.found || [];
+
+    const mlsConversations = foundConversations.filter(isMLSConversation);
+
+    try {
+      //check all the established conversations' epoch with the core-crypto epoch
+      for (const {qualified_id: qualifiedId, group_id: groupId, epoch} of mlsConversations) {
+        const isEstablished = await this.isMLSConversationEstablished(groupId);
+        if (!isEstablished) {
+          this.logger.info(
+            `MLS conversation with id ${qualifiedId.id} was found in the database, but not established yet, joining via external commit`,
+          );
+          //if conversation was found bot not yet established, join
+          await this.joinByExternalCommit(qualifiedId);
+          continue;
+        }
```

Comment:
I wonder if we could simplify the logic a tiny bit. Since we always want to call `joinByExternalCommit` whether the conversation is a mismatch OR is not established. 

I'm wondering: what does `getEpoch` returns if the conversation is not established? Does it return some special value (like -1) or does it throw?

Reply:
Good point, I'll check it out 👍

Reply:
we could probably do something along those lines

```js
if (!isEstablished(conv) || isEpochMismatch(conv, newEpoch) {
  joinByExternal(conv)
}
```
WDYT?

Reply:
It seems that `getEpoch` throws an error when it cannot find conversation with given `groupId`